virtwho/executor.py

```diff
@@ -68,8 +74,36 @@ def _create_virt_backends(self):
                 self.logger.error('Unable to use configuration "%s": %s', name, str(e))
                 continue
             virts.append(virt)
+        self._init_run_status(config_names)
         return virts
 
+    def _init_run_status(self, config_names=[]):
+        # ensure that there is an entry in the status file for each config. We don't care if they work, we
+        # need to record their existence
+        lock = PIDLock(STATUS_LOCK)
+        while lock.is_locked():
+            time.sleep(1)
```

Comment:
I think that this waiting while loop should not be here. When lock is locked, then following `with lock` is blocked until `lock` is unblocked.

Reply:
Turns out that the original use of the PIDLock was only to stop from running more than one virt-who process. It did not wait to get the lock, but just reported it and stopped the process. 
I put the wait loop into the PIDLock class and made it conditional so as to not disrupt the original execution.